providers/unleash/src/test/java/dev/openfeature/contrib/providers/unleash/UnleashProviderIntegrationTest.java

```diff
@@ -0,0 +1,94 @@
+package dev.openfeature.contrib.providers.unleash;
+
+import dev.openfeature.sdk.Client;
+import dev.openfeature.sdk.ImmutableContext;
+import dev.openfeature.sdk.OpenFeatureAPI;
+import io.getunleash.strategy.DefaultStrategy;
+import io.getunleash.strategy.Strategy;
+import io.getunleash.strategy.UserWithIdStrategy;
+import io.getunleash.util.UnleashConfig;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * UnleashProvider Integration test via live Unleash instance.
+ * To trigger manually only.
+ * To test it, set API_KEY and other values accordingly.
+ */
+@Slf4j
+class UnleashProviderIntegrationTest {
+
+    private static final String FLAG_NAME = "open-feature_flag1";
+    public static final String API_KEY = null;
+    public static final String API_URL = "https://app.unleash-hosted.com/demo/api/";
+    public static final String APP_NAME = "open-feature";
```

Comment:
I wonder if we can use docker containers for these tests. It should be easy both for dev and for CI: https://github.com/Unleash/unleash-server-api-go/blob/main/.github/workflows/build.yml#L30-L31

In particular, you can fix the admin token so it's known at runtime: https://github.com/Unleash/unleash-server-api-go/blob/fcd6e223b3a6c33a9e8394c5fbcf27f5995277e3/docker-compose.yml#L19-L31

Reply:
Ya I would recommend a dockerized integration test, and (if possible) that you leverage the standard integration test suite defined in the spec: https://github.com/open-feature/spec/blob/main/specification/assets/gherkin/evaluation.feature

You can omit specific assertions (such as reason) as your provider requires, but this covers a lot of surface area, including context parsing, object parsing, type mapping and error handling.

Reply:
I don't think this is as important with the way @liran2000 has done it. Mocking HTTP responses seems like a pretty good solution, as long as the mocked response is accurate these tests should be pretty solid.